lib/validation/validators/requiredHmsIf.ts

```diff
@@ -0,0 +1,19 @@
+import { type HmsType, isHms } from '../../support/Day'
+import { requiredHms } from './requiredHms'
+import { type RequiredIfCondition } from './requiredIf'
+
+export async function requiredHmsIf(value: unknown, condition: RequiredIfCondition, required: HmsType[] = ['h', 'm', 's']): Promise<boolean> {
+  if (typeof condition === 'boolean' && condition) {
+    return requiredHms(value, required)
+  }
+
+  if (typeof condition === 'string' && condition) {
+    return requiredHms(value, required)
+  }
+
+  if (typeof condition === 'function' && (await condition())) {
+    return requiredHms(value, required)
+  }
+
+  return isHms(value, required)
```

Comment:
This should be true I guess 🤔 Otherwise, the validator might return false even if the condition is not met. When condition doesn't meet, it should not validate anything at all.

Reply:
Yeah this validator will fail if value is nullish and condition is falsy. The value shouldn't be required if condition is not true / doesn't resolve to true.

Reply:
Fixed: 21c1168